### DISCUSSION TITLE
Help with understanding how to work with react-hook-form

### DISCUSSION THREAD

**Discussion by santafecap** [shadcn-ui/ui#3726] [Q&A]
Hello

I haven't even heard of react-hook-form until starting to use shadcn-ui so please forgive my understanding. 

I created a form based off dashboard-03 blocks, specifically there is this starting part:

```
 <form className="grid w-full items-start gap-6">
   <fieldset className="grid gap-6 rounded-lg border p-4">
```
which as I copied into my project, I was getting the error outlined in https://github.com/shadcn-ui/ui/issues/800 however I did not import 'Form' from react-hook-form. the only way I can resolve the error is with:

```
const form = useForm();

<Form {...form}>
 <form className="grid w-full items-start gap-6">
  <fieldset className="grid gap-6 rounded-lg border p-4">
```
now what is extremely bothering me is that, I added more variables from useForm() which then produced the previous errors again, and incredulously the only way to solve it is as follows:

```
const form = useForm();
const {register, watch} = useForm();
```
while 

`const {form, register, watch} = useForm();` Does not work!! 

how could this be that you have to separately declare `form`?? I am not finding anything that explains this behavior on react-hook-form docs. 

Any insights would be appreciated.

**Comment by Maraco91** [comment]
Hi, try this

```typescript
const form = useForm();
const { watch } = form
// Example
const watchName= watch("name")

useEffect(() => {
   console.log(watchName)
}, [watchName])
```
